inspire/utils/latex.py

```diff
@@ -0,0 +1,428 @@
+                    return ''
+            if isinstance(citation_key, list):
+                for element in citation_key:
+                    return element.replace(' ', '')
+            else:
+                return citation_key.replace(' ', '')
+        else:
+            return ''
+
+    def _fetch_fields(self, req_fields, opt_fields=[]):
+        fields = {
+            'author': self._get_author,
+            'title': self._get_title,
+            'publi_info': self._get_publi_info,
+            'arxiv': self._get_arxiv,
+            'report_number': self._get_report_number,
+            'SLACcitation': self._get_slac_citation,
+        }
+        out = ''
+        for field in req_fields:
+            value = fields[field]()
+            if value:
+                out += self._format_output_row(field, value)
+            # RAISE EXCEPTION HERE IF REQ FIELD IS MISSING
+        for field in opt_fields:
+            value = fields[field]()
+            if value:
+                out += self._format_output_row(field, value)
+        return out
+
+    def _format_output_row(self, field, value):
+        out = ''
+        if field == 'author':
+            if len(value) == 1:
+                out += u'  {1},\n'.format(field, value[0])
+            elif len(value) > 8:
+                out += u'  {1}'.format(field, value[0])
+                if 'collaboration' in self.record:
+                    try:
+                        if 'collaboration' in self.record['collaboration'][0]:
+                            collaboration = self.\
+                                    record['collaboration'][0]['collaboration']
+                            if 'Collaboration' in collaboration:
+                                out += ' {{\it et al.}} [' + collaboration + \
+                                   '],\n'
+                            else:
+                                out += ' {{\it et al.}} [' + collaboration + \
+                                   ' Collaboration],\n'
+                    except IndexError:
+                        pass
+                else:
+                    out += u' {\it et al.},\n'
+            else:
+                out += '  {} and {},\n'.format(', '.join(value[:-1]),
+                                               value[-1])
+        elif field == 'title':
+            out += u'  {1}\n'.format(field, value)
+        elif field == 'publi_info':
+            out += u'  {1}.\n'.format(field, value)
+        elif field == 'arxiv':
+            if self._get_publi_info():
+                out += u'  [{1}].\n'.format(field, value)
+            else:
+                out += u'  {1}.\n'.format(field, value)
+        elif field == 'report_number':
+            out += u'  {1}.\n'.format(field, value)
+        elif field == 'SLACcitation':
+            out += u'  {1}'.format(field, value)
+        return out
+
+    def _get_arxiv_field(self):
+        """Return arXiv field if exists"""
+        if 'report_number' in self  .record:
```

Comment:
That might happened accidentally, didn't see it, strange that the pep8
didn't complain :/

2015-08-26 9:51 GMT+02:00 Javier Martin Montull <notifications@github.com>:

> In inspire/utils/latex.py
> <https://github.com/inspirehep/inspire-next/pull/320#discussion_r37954502>
> :
>
> > +        elif field == 'publi_info':
> > +            out += u'  {1}.\n'.format(field, value)
> > +        elif field == 'arxiv':
> > +            if self._get_publi_info():
> > +                out += u'  [{1}].\n'.format(field, value)
> > +            else:
> > +                out += u'  {1}.\n'.format(field, value)
> > +        elif field == 'report_number':
> > +            out += u'  {1}.\n'.format(field, value)
> > +        elif field == 'SLACcitation':
> > +            out += u'  {1}'.format(field, value)
> > +        return out
> > +
> > +    def _get_arxiv_field(self):
> > +        """Return arXiv field if exists"""
> > +        if 'report_number' in self  .record:
>
> Is there an space in self .record ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/inspirehep/inspire-next/pull/320/files#r37954502>.
>